pennylane_lightning/lightning_kokkos/lightning_kokkos.py

```diff
@@ -614,6 +614,9 @@ def generate_samples(self, shots=None):
             ``(dev.shots, dev.num_wires)``
         """
         shots = self.shots if shots is None else shots
+
+        shots = shots if isinstance(shots, int) else shots.total_shots
```

Comment:
It'd be a better practice to check the instance before accessing the attribute:  
```suggestion
        shots = shots.total_shots if isinstance(shots, qml.measurements.Shots) else shots
```